docs/angular/rxobject.md

```diff
@@ -0,0 +1,416 @@
+
+This distinction is extremely important for the purposes of our project. We want to use subjects so that multiple components can view the same stream of data!
+
+There are three different kinds of `Subject<T>`s that we can choose from. All of these three specific types *extend* `Subject<T>` adding new functionality:
+
+- `BehaviorSubject<T>`: A variant of `Subject<T>` that *requires an initial value* and *emits its current value whenever it is subscribed to*.
+- `ReplaySubject<T>`: A variant of `Subject<T>` that *"replays" old values* to new subscribers *by emitting them when they first subscribe*.
+- `AsyncSubject<T>`: A variant of `Subject<T>` that only emits a value when it completes.
+
+For the purposes of being able to update our data dynamically and share it across components, we are going to take a closer look at the `ReplaySubject<T>` class.
+
+### The `ReplaySubject<T>` Class
+
+As noted in the section above, the `ReplaySubject<T>` re-emits the latest value that is has to *new subscribers*. This is extremely useful. Imagine that we loaded all organization data on the main organization page, and then navigate to a new page that uses the same data. By re-subscribing to the `ReplaySubject<T>`, we can essentially get access to the most recently-loaded data so that we do not have to re-fetch all of our data again!
+
+We can create a `ReplaySubject<T>` object using the code below:
+
+```ts
+let subject: Subject<string> = new ReplaySubject(1);
+```
+
+> **NOTE:** The subject created above stores *strings* as its value. However, you can replace this data type with any data type you like.
+
+It is important to note that the `ReplaySubject()` constructor takes a *parameter*. This parameter is the ***buffer size***, or the amount of past data that it should store! In this case, we only want to keep the latest (most up-to-date) data in our subject, so we can keep our buffer size as `1`.
+
+Once we created the `ReplaySubject` object, we can then *feed data into it* (set its value) using the `.next()` method. This method also ***notifies*** all observers pointing to the subject.
+
+```ts
+let subject: Subject<string> = new ReplaySubject(1);
+subject.next("COMP 423 rocks!!");
+```
+
+The `.next()` method is how we update data within our subject!
+
+It is also important to note that we can *convert subjects back to observers*. For example, given the same subject we created before, we can create a typical `Observable<T>` object out of it using the following:
+
+```ts
+let subject: Subject<string> = new ReplaySubject(1);
+let value$: Observable<string> = subject.asObservable();
+
+// Access latest data from the subject
+value$.subscribe((stringValue) => {
+  console.log(stringValue);
+})
+```
+
+As you can see, the code above creates an `Observable` from the `ReplaySubject` which we can subscribe to!
```

Comment:
This is where it can be pointed out the key distinction -- new values can be pushed to `subject`, but `value$` is read-only.